File: account/regional-settings/get-states-list/code-samples/listStates.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# OPTIONAL QUERY PARAMETERS
queryParams = {
    #'allCountries': true,
    #'countryId': 000,
    #'page': 1,
    #'perPage': 100,
    #'withPhoneNumbers': true
}

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])
r = platform.get('/restapi/v1.0/dictionary/state', queryParams)

File: account/regional-settings/get-states-list/code-samples/listStates.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //allCountries: true,
    //countryId: 000,
    //page: 1,
    //perPage: 100,
    //withPhoneNumbers: true
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.get(`/restapi/v1.0/dictionary/state`, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});
